browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc

```diff
@@ -226,15 +226,32 @@ void AIChatUIPageHandler::GetAPIResponseError(
   std::move(callback).Run(active_chat_tab_helper_->GetCurrentAPIError());
 }
 
-void AIChatUIPageHandler::GetHasUserDismissedPremiumPrompt(
-    GetHasUserDismissedPremiumPromptCallback callback) {
-  std::move(callback).Run(profile_->GetPrefs()->GetBoolean(
-      ai_chat::prefs::kUserDismissedPremiumPrompt));
+void AIChatUIPageHandler::MaybeShowPremiumPrompt(
+    MaybeShowPremiumPromptCallback callback) {
+  bool has_user_dismissed = profile_->GetPrefs()->GetBoolean(
+      ai_chat::prefs::kUserDismissedPremiumPrompt);
+
+  if (has_user_dismissed) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  base::Time last_seen_disclaimer =
+      profile_->GetPrefs()->GetTime(ai_chat::prefs::kBravekLastSeenDisclaimer);
+  base::Time time_1_day_ago = base::Time::Now() - base::Days(1);
```

Comment:
Just a note that we can add to the test plan that the tester can forward the clock or manually change this pref value to force this to happen.

Reply:
I don't think this logic belongs in the page handler. Don't we want to do something similar on android/ios?